module3/homework/inner_product/arithmeticAverage.cpp

```diff
@@ -0,0 +1,62 @@
+#include "arithmeticAverage.hpp"
+
+#include <algorithm>
+#include <functional>
+#include <math.h>
+#include <numeric>
+#include <stdexcept>
+#include <vector>
+
+double ArithmeticAverage(std::vector<int> first, std::vector<int> second) {
+    std::vector<int> longer;
+    std::vector<int> shorterOrEqual;
+
+    if (first.size() == 0 && second.size() == 0) {
+        throw std::invalid_argument("At least one vector should not be empty");
+    }
+    else if (first.size() == 0) {
+        longer.reserve(second.size());
+        longer = second;
+        shorterOrEqual.reserve(second.size());
+        shorterOrEqual = second;
+    }
+    else if (second.size() == 0) {
+        longer.reserve(first.size());
+        longer = first;
+        shorterOrEqual.reserve(first.size());
+        shorterOrEqual = first;
+    }
+    else if (first.size() >= second.size()) {
+        longer.reserve(first.size());
+        longer = first;
+        shorterOrEqual.reserve(second.size());
+        shorterOrEqual = second;
+    }
+    else {
+        longer.reserve(second.size());
+        longer = second;
+        shorterOrEqual.reserve(first.size());
+        shorterOrEqual = first;
+    }
```

Comment:
I'm not sure if I have understood innerProduct task properly,but I thought that both vector should have the same size.

Reply:
In my opinion both vectors dont have to be the same size. For example. Vec1 {1, 2, 3}, Vec2 {1, 2}. 
We can still calculate the arithmetic mean. Inner_product will sum '3' with the default value with is '0'.

Reply:
I've changed the code to demand vectors to be the same size. But at first I wanted to make it work no matter what is the size of any vector. Although based on possible implementation from https://en.cppreference.com/w/cpp/algorithm/inner_product it seems that when second vector is shorter then the second iterator can go further out of bound and get some random values from memory.